src/test/java/org/cyclonedx/BomXmlGeneratorTest.java

```diff
@@ -79,9 +96,11 @@ public void schema11GenerationTest() throws Exception {
     }
 
     @Test
-    public void schema11WithDependencyGraphGenerationTest() throws Exception {
-        BomXmlGenerator generator = BomGeneratorFactory.createXml(CycloneDxSchema.Version.VERSION_11, createCommonBom("/bom-1.1-dependency-graph-1.0.xml"));
-        generator.generate();
+    public void schema11GenerationTest() throws Exception {
+        BomXmlGenerator generator = BomGeneratorFactory.createXml(CycloneDxSchema.Version.VERSION_11, createCommonBom());
+        Document doc = generator.generate();
+        testDocument(doc);
+
         Assert.assertTrue(generator instanceof BomXmlGenerator11);
         Assert.assertEquals(CycloneDxSchema.Version.VERSION_11, generator.getSchemaVersion());
         File file = writeToFile(generator.toXmlString());
```

Comment:
This test (`schema12GenerationTest()`) is failing for me, with the following exception in the `XmlParser.validate()` method:
```
cvc-complex-type.2.4.a: Invalid content was found starting with element '{"http://cyclonedx.org/schema/bom/1.2":url}'. One of '{"http://cyclonedx.org/schema/bom/1.2":contact, WC[##other:"http://cyclonedx.org/schema/bom/1.2"]}' is expected.

Reply:
Well crud. I'm sure I had done a `git pull`, but re-re-running the test suite, now it's passing. Please ignore my comment while I scratch my head and try to figure out what I did wrong.

Reply:
@DarthHater, @stevespringett  - BTW, I think the above intermittent failure is due to the `.xsd` defining some sub tags using `<xs:sequence ...`, which requires the sub tags be in a specific order. Might be worth using
 `<xsd:sequence>      
      <xsd:choice ...` to permit  any of these tags in any order.

Reply:
@darthhater, @stevespringett FWIW, this patch to the `.xsd` seems to solve the intermittent failures involving an `organizationalEntity`.

```
Index: src/main/resources/bom-1.2.xsd
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/bom-1.2.xsd b/src/main/resources/bom-1.2.xsd
--- a/src/main/resources/bom-1.2.xsd	(revision 06ea2769a786d99d44a3015b028f2b1dad31e1ae)
+++ b/src/main/resources/bom-1.2.xsd	(date 1609865728327)
@@ -99,29 +99,31 @@
     </xs:complexType>
 
     <xs:complexType name="organizationalEntity">
-        <xs:sequence minOccurs="0" maxOccurs="1">
-            <xs:element name="name" type="xs:normalizedString" minOccurs="0" maxOccurs="1">
-                <xs:annotation>
-                    <xs:documentation>The name of the organization</xs:documentation>
-                </xs:annotation>
-            </xs:element>
-            <xs:element name="url" type="xs:anyURI" minOccurs="0" maxOccurs="unbounded">
-                <xs:annotation>
-                    <xs:documentation>The URL of the organization. Multiple URLs are allowed.</xs:documentation>
-                </xs:annotation>
-            </xs:element>
-            <xs:element name="contact" type="bom:organizationalContact" minOccurs="0" maxOccurs="unbounded">
-                <xs:annotation>
-                    <xs:documentation>A contact person at the organization. Multiple contacts are allowed.</xs:documentation>
-                </xs:annotation>
-            </xs:element>
-            <xs:any namespace="##other" processContents="lax" minOccurs="0" maxOccurs="unbounded">
-                <xs:annotation>
-                    <xs:documentation>
-                        Allows any undeclared elements as long as the elements are placed in a different namespace.
-                    </xs:documentation>
-                </xs:annotation>
-            </xs:any>
+        <xs:sequence>
+            <xs:choice minOccurs="0" maxOccurs="unbounded">
+                <xs:element name="name" type="xs:normalizedString" minOccurs="0" maxOccurs="1">
+                    <xs:annotation>
+                        <xs:documentation>The name of the organization</xs:documentation>
+                    </xs:annotation>
+                </xs:element>
+                <xs:element name="url" type="xs:anyURI" minOccurs="0" maxOccurs="unbounded">
+                    <xs:annotation>
+                        <xs:documentation>The URL of the organization. Multiple URLs are allowed.</xs:documentation>
+                    </xs:annotation>
+                </xs:element>
+                <xs:element name="contact" type="bom:organizationalContact" minOccurs="0" maxOccurs="unbounded">
+                    <xs:annotation>
+                        <xs:documentation>A contact person at the organization. Multiple contacts are allowed.</xs:documentation>
+                    </xs:annotation>
+                </xs:element>
+                <xs:any namespace="##other" processContents="lax" minOccurs="0" maxOccurs="unbounded">
+                    <xs:annotation>
+                        <xs:documentation>
+                            Allows any undeclared elements as long as the elements are placed in a different namespace.
+                        </xs:documentation>
+                    </xs:annotation>
+                </xs:any>
+            </xs:choice>
         </xs:sequence>
         <xs:anyAttribute namespace="##other" processContents="lax">
             <xs:annotation>
```

Reply:
I am just closing the loop here, the order is intentional, @stevespringett had told me that, and I think we just gotta make sure that the properties always generate in the right order.

Reply:
The properties generating in random order is likely something having to do with how it gets the properties via Reflection. I noticed at least when I was experimenting with `xstream` that in some JDKs they came back ordered, in others they did not. Very peculiar and odd, and I believe influences how a few libraries construct JSON/XML using a POJO as a blueprint.

Reply:
Did something about this in: 
cedeb1b